File: src/main/java/com/core/base/corebase/common/dto/BaseExceptionRes.java
package com.core.base.corebase.common.dto;

import com.core.base.corebase.common.code.ErrorCode;
import lombok.Getter;

import java.time.ZonedDateTime;

@Getter
public class BaseExceptionRes {
    private final ErrorCode code;
    private final String reason;
    private final int status;
    private final String timeStamp;

    public BaseExceptionRes(ErrorCode code, String reason) {
        this.code = code;
        this.reason = reason;
        this.status = code.getCode().value();
        this.timeStamp = ZonedDateTime.now().toString();
    }
}

File: src/main/kotlin/com/core/base/corebase/common/dto/BaseExceptionRes.kt
package com.core.base.corebase.common.dto

import com.core.base.corebase.common.code.ErrorCode
import java.time.ZonedDateTime

class BaseExceptionRes(
    val code: ErrorCode,
    val reason : String?
) {
    val status = code.code
    val timeStamp = ZonedDateTime.now().toString()
}